File: ios/VideoPlayerViewManager.m
// VideoPlayerViewManager.m
#import <React/RCTViewManager.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTConvert.h>

@interface RCT_EXTERN_MODULE(VideoPlayerViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(videoURL, NSString)
RCT_EXPORT_VIEW_PROPERTY(borderRadius, CGFloat)
RCT_EXPORT_VIEW_PROPERTY(resizeMode, NSString)

// Event callbacks
RCT_EXPORT_VIEW_PROPERTY(onProgress, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onEnd, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onLoad, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onError, RCTDirectEventBlock)

// Register the commands with RCT_EXTERN_METHOD only
RCT_EXTERN_METHOD(play:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(pause:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(seekTo:(nonnull NSNumber *)node time:(nonnull NSNumber *)time)

@end

File: ios/VideoPlayerViewManager.swift
//
//  VideoPlayerViewManager.swift
//  quasarcommunity
//
//  Created by Quasar1 on 2/11/25.
//

import Foundation
import React

@objc(VideoPlayerViewManager)
class VideoPlayerViewManager: RCTViewManager {
  override func view() -> UIView! {
    return VideoPlayerView()
  }
  
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc func play(_ node: NSNumber) {
    print("Play command received for node:", node)
    DispatchQueue.main.async {
      guard let component = self.bridge.uiManager.view(forReactTag: node) as? VideoPlayerView else {
        print("Failed to find VideoPlayerView for node:", node)
        return
      }
      print("Found VideoPlayerView, calling play")
      component.play()
    }
  }
  
  @objc func pause(_ node: NSNumber) {
    print("Pause command received for node:", node)
    DispatchQueue.main.async {
      guard let component = self.bridge.uiManager.view(forReactTag: node) as? VideoPlayerView else {
        print("Failed to find VideoPlayerView for node:", node)
        return
      }
      print("Found VideoPlayerView, calling pause")
      component.pause()
    }
  }
  
  @objc func seekTo(_ node: NSNumber, time: NSNumber) {
    print("Seek command received for node:", node, "time:", time)
    DispatchQueue.main.async {
      guard let component = self.bridge.uiManager.view(forReactTag: node) as? VideoPlayerView else {
        print("Failed to find VideoPlayerView for node:", node)
        return
      }
      print("Found VideoPlayerView, calling seekTo")
      component.seekTo(time: time.floatValue)
    }
  }
}
